File: C codes/selectionSort.c
#include <stdio.h>
void swap(int x, int y)
{
    int temp = x;
    x = y;
    y = temp;
}
 void printArray(int arr[], int size)
{
    int i;
    for (i=0; i < size; i++)
        printf("%d ", arr[i]);
    printf("\n");
}
void selectionSort(int arr[], int n)
{
    int i, j, min;
    for (i = 0; i < n-1; i++)
    {
        min = i;
        for (j = i+1; j < n; j++)
          if (arr[j] < arr[min])
            min = j;
        swap(arr[min], arr[i]);
    }
    printArray(arr,n);
}

File: arrays/Sorting/SelectionSort.java
package arrays.Sorting;

import java.util.Scanner;

public class SelectionSort {

    public int[] selectionSort(int[] array){
        int min;
        for(int i = 0; i < array.length - 1; i++){
            min = i;
            for(int j =  i + 1; j < array.length; j++){
                if(array[j] < array[min]){
                    min = j;
                }
            }
            if(min != i){
                int temp = array[i];
                array[i] = array[min];
                array[min] = temp;
            }
        }

        return array;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        SelectionSort sort = new SelectionSort();
        int size = sc.nextInt();
        int[] array = new int[size];

        for(int i = 0; i < array.length; i++){
            array[i] = sc.nextInt();
        }

        array = sort.selectionSort(array);

        for(int i : array){
            System.out.print( i + " ");
        }
    }
}
